pkg/transport/generic_transport_controller.go

```diff
@@ -0,0 +1,512 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transport
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cacheddiscovery "k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/dynamicinformer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"github.com/kubestellar/kubestellar/api/edge/v1alpha1"
+	edgeclientset "github.com/kubestellar/kubestellar/pkg/generated/clientset/versioned"
+	edgev1alpha1informers "github.com/kubestellar/kubestellar/pkg/generated/informers/externalversions/edge/v1alpha1"
+	edgev1alpha1listers "github.com/kubestellar/kubestellar/pkg/generated/listers/edge/v1alpha1"
+)
+
+const (
+	ControllerName                 = "transport-controller"
+	transportFinalizer             = "transport.kubestellar.io/object-cleanup"
+	notFoundErrorSuffix            = "not found"
+	originOwnerReferenceAnnotation = "transport.kubestellar.io/originOwnerReferencePlacementDecisionKey"
+	originWdsAnnotation            = "transport.kubestellar.io/originWdsName"
+)
+
+// NewTransportController returns a new transport controller
+func NewTransportController(ctx context.Context, placementDecisionInformer edgev1alpha1informers.PlacementDecisionInformer, transport Transport,
```

Comment:
FYI, Kubernetes has established confused terminology:
```
type PlacementDecisionInformer interface {
	Informer() cache.SharedIndexInformer
	Lister() v1alpha1.PlacementDecisionLister
}
```

Thus, one "informer" of PlacementDecisions has a method `Informer` that produces another "informer" of PlacementDecisions (alongside a method that produces a lister).

FYI, I tend to name variables of the former type "...PreInformer" rather than "Informer", to make the distinction clearer.

Reply:
yes. I'm totally aware of this.
the names I used are following the patterns from the community sample controller:
https://github.com/kubernetes/sample-controller/blob/master/controller.go#L94

whoever is reading this go-client code should be able to understand it.

Reply:
Indeed it's confusing. I'm not sure PreXX is helping here (Personally I was always confused by Pre - I thought it performs operations prior to those of the informer method )
It's more like InformerFactory  but probably not that as well.   
I'm fine leaving it like this unless we comup with a good naming convention . @pdettori ?

Reply:
I'm not sure I understand why do we need to come up with new naming convention while using `informer` IS THE naming convention in the community.
look for your favorite controller in k8s org, this is the pattern:
https://github.com/kubernetes/kubernetes/blob/81986587efb506c54632babf0ec79592da79597a/pkg/controller/volume/persistentvolume/pv_controller_base.go#L70-#L74

Reply:
The value is "pre" in the sense that you get this value _before_ you get the actual informer.

It is OK with me to leave this code as it is. I was just suggesting something that I find helpful. I do not think we have to follow examples in all details. Remember that the code upstream has evolved over time, and has a lot of warts due to its history and the fact that it evolved by changing only what _had_ to be changed and got limited review.